src/nodes/limiter.rs

```diff
@@ -0,0 +1,381 @@
+struct AsymmetricalSmoothedParam {
+    target_value: f32,
+    target_times_a_up: f32,
+    target_times_a_down: f32,
+    filter: SmoothingFilter,
+    coeff_up: SmoothingFilterCoeff,
+    coeff_down: SmoothingFilterCoeff,
+    smooth_secs_up: f32,
+    smooth_secs_down: f32,
+}
+
+impl AsymmetricalSmoothedParam {
+    /// Construct a new smoothed f32 parameter with the given configuration.
+    pub fn new(value: f32, config: AsymmetricalSmootherConfig, sample_rate: NonZeroU32) -> Self {
+        assert!(config.smooth_secs_up > 0.0);
+        assert!(config.smooth_secs_down > 0.0);
+
+        let coeff_up = SmoothingFilterCoeff::new(sample_rate, config.smooth_secs_up);
+        let coeff_down = SmoothingFilterCoeff::new(sample_rate, config.smooth_secs_down);
+
+        Self {
+            target_value: value,
+            target_times_a_up: value * coeff_up.a0,
+            target_times_a_down: value * coeff_down.a0,
+            filter: SmoothingFilter::new(value),
+            coeff_up,
+            coeff_down,
+            smooth_secs_up: config.smooth_secs_up,
+            smooth_secs_down: config.smooth_secs_down,
+        }
+    }
+
+    /// The target value of the parameter.
+    pub fn target_value(&self) -> f32 {
+        self.target_value
+    }
+
+    /// Set the target value of the parameter.
+    pub fn set_value(&mut self, value: f32) {
+        self.target_value = value;
+        self.target_times_a_up = value * self.coeff_up.a0;
+        self.target_times_a_down = value * self.coeff_down.a0;
+    }
+
+    pub fn set_smooth_secs_up(&mut self, sample_rate: NonZeroU32, smooth_secs_up: f32) {
+        let coeff_up = SmoothingFilterCoeff::new(sample_rate, smooth_secs_up);
+        self.smooth_secs_up = smooth_secs_up;
+        self.coeff_up = coeff_up;
+    }
+
+    pub fn set_smooth_secs_down(&mut self, sample_rate: NonZeroU32, smooth_secs_down: f32) {
+        let coeff_down = SmoothingFilterCoeff::new(sample_rate, smooth_secs_down);
+        self.smooth_secs_down = smooth_secs_down;
+        self.coeff_down = coeff_down;
+    }
+
+    /// Return the next smoothed value.
+    #[inline(always)]
+    pub fn next_smoothed(&mut self) -> f32 {
+        if self.filter.z1 < self.target_value() {
+            self.filter
+                .process_sample_a(self.target_times_a_up, self.coeff_up.b1)
+        } else {
+            self.filter
+                .process_sample_a(self.target_times_a_down, self.coeff_down.b1)
+        }
+    }
+
+    /// Update the sample rate.
+    pub fn update_sample_rate(&mut self, sample_rate: NonZeroU32) {
+        self.coeff_up = SmoothingFilterCoeff::new(sample_rate, self.smooth_secs_up);
+        self.coeff_down = SmoothingFilterCoeff::new(sample_rate, self.smooth_secs_down);
+        self.target_times_a_up = self.target_value() * self.coeff_up.a0;
+        self.target_times_a_down = self.target_value() * self.coeff_down.a0;
+    }
+}
+
+/// Buffer.
+#[derive(Debug, Clone)]
+pub struct IncrementalMax {
```

Comment:
Should this be public? If so, let's expand on the docs a bit. "Buffer" is uh... a little concise!